File: Fights/chessKnightMoves.js
function chessKnightMoves(cell) {
    x = cell.charCodeAt(0)%97;
    y = cell.charCodeAt(1) - '0'.charCodeAt(0) - 1;
    c = 0;
    for ( dx = -2; dx <= 2; dx++)
        for (dy = -2; dy <= 2; dy++) {
            if (Math.abs(dx * dy) == 2)
                if (isValid(x + dx, y + dy))
                    c++;
        }
    return c;
}
function isValid(x, y) {
    return x >= 0 && x <= 7 && y >= 0 && y <= 7;
}

File: Fights/chessKnightMoves.py
def chessKnightMoves(cell):
    x=ord(cell[0])-97
    y=ord(cell[1])-49
    c=0
    for i in range(-2,3):
        for j in range(-2,3):
            if abs(i*j)==2:
                if (check(x+i,y+j)):
                    c+=1
    return c
def check(x,y):
    return x >= 0 and x <= 7 and y >= 0 and y <= 7
